backend/src/models/giftTransfer.model.ts

```diff
@@ -61,7 +63,7 @@ giftTransferSchema.index({
   userId: 'text'
 })
 
-giftTransferSchema.pre('save', async function () {
+giftTransferSchema.pre('save', async function() {
   const currentDocument = this as IGiftTransferDocument
```

Comment:
https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#specifying-the-type-of-this-for-functions